### ISSUE TITLE
[Training] model architecture related question

### ISSUE THREAD

**Issue by daiyingjie2024** [KAIST-AILab/SyncVSR#11]
Sorry, I didn't consider the different categories of the issues earlier. I should have opened separate issues.Do you think the architecture for lip reading tasks has already become standardized? I might not have read many papers, but it seems like the visual front end is mostly 3D-ResNet, and the sequence back end is usually Conformer or TCN.

**Comment by snoop2head** [comment]
@daiyingjie2024 

Thank you for filing as a separate issue!

We deliberately opted for the legacy architecture to control for potential exogenous variables aside from our audio reconstruction loss term.

### Temporal Backend

You're absolutely right! Many existing works tend to use either Conformer or TCN as the temporal backend. I assume this preference is due to the inherent inductive biases of CNN layers, such as translation equivariance and locality, which we'll revisit later.

As illustrated in the diagram below, the Transformer model (purple) initially underperforms compared to the TCN model (orange) when trained using our audio reconstruction loss term. However, as the loss coefficient for our term increases, the Transformer model begins to outperform the TCN model.

<img width="450" alt="image" src="https://github.com/KAIST-AILab/SyncVSR/blob/main/assets/lambda.png?raw=true">

I believe this is because that our loss term shifts the bias of temporal encoder's the inner representation towards local neighboring frames, as we have illustrated in the [Figure 4 of our paper](https://arxiv.org/abs/2406.12233). We believe that training with our framework makes encoders to provide following classifiers (or autoregressive decoders) with better representations, allowing them to distinguish fine-grained visemes based on encoder’s representation. As shown in the diagram above, we believe that improvements from our audio reconstruction loss term is orthogonal to that of the architectural approaches, where you might consider training Conformer with our audio reconstruction term, which we believe is promising.

### Visual frontend
The use of Conv3D in the visual frontend can bottleneck the model’s throughput, which can be replaced with a simple linear projection as we can see in [LP-Conformer](https://arxiv.org/abs/2302.10915). 

However, this replacement is only viable when the training dataset is sufficiently large. LP-Conformer, for example, is trained on a dataset comprising 100K hours of video. As the authors of the [ViT paper](https://arxiv.org/pdf/2010.11929) state:

> Transformers lack some of the inductive biases inherent to CNNs, such as translation equivariance and locality, and therefore do not generalize well when trained on insufficient amounts of data ... We find that large scale training trumps inductive bias.

In our case of training on landmark-based VSR, we used Conv1D window size of 1 frame and a stride of 1 frame (=linear projection) rather than Conv3D. This decision was based on the assumption that the LRW dataset provided sufficient data relative to the size of the model to allow for generalization. However, we had to train the model for over 1000 epochs with heavy augmentation to achieve this.

https://github.com/KAIST-AILab/SyncVSR/blob/8a99b28acef5451073f68f3fd64749b7ceffd3b4/LRW/landmark/src/modeling.py#L113-L119

The choice of model design is ultimately up to individual researchers, but I recommend carefully considering the size of the dataset you are using for training when making these decisions.

**Comment by davidingram123** [comment]
@snoop2head I’m not sure if I have misunderstood you, but what you’re saying is:  the audio reconstruction loss term proposed in the paper causes the temporal backend to be trained to pay more attention to the relationship between each frame and its adjacent frames, such that, even without the inductive biases provided by CNNs, the transformer can still perform relatively well?

**Comment by snoop2head** [comment]
@davidingram123 
Yeap! That was my interpretation based on how Transformer model surpasses the TCN backbones based on our audio reconstruction loss.

**Comment by davidingram123** [comment]
@snoop2head Thank you, I understand.
